.github/workflows/ci.yml

```diff
@@ -0,0 +1,128 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - '*'
+      - '!*backport*'
+    tags:
+      - 'v*'
+      - '!*dev*'
+      - '!*pre*'
+      - '!*post*'
+  pull_request:
+  # Allow manual runs through the web UI
+  workflow_dispatch:
+
+jobs:
+  core:
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      libraries: |
+        apt:
+          - libopenjp2-7
+      envs: |
+        - linux: py39
+
+  test:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto
+      libraries: |
+        apt:
+          - libopenjp2-7
+        brew:
+          - openjpeg
+      envs: |
+        - macos: py38
+        - windows: py310
+        - linux: py38-oldestdeps
+        - linux: py39-conda
+          libraries: ''
+
+  online:
+    needs: [core]
+    uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
+    with:
+      default_python: '3.8'
+      submodules: false
+      coverage: codecov
+      toxdeps: tox-pypi-filter
+      posargs: -n auto --dist loadgroup
+      libraries: |
+        apt:
+          - libopenjp2-7
+      envs: |
+        - linux: build_docs_ci
+          posargs: ''
+          pytest: false
+          libraries:
+            apt:
+              - libopenjp2-7
+              - graphviz
+        - linux: py38-online
+
+  cron:
```

Comment:
Same with the cron-only builds, I think they should be in their own workflow. We can also copy [this trick](https://github.com/astropy/astropy/blob/main/.github/workflows/ci_cron_daily.yml#L30) I came up with for Astropy where by adding the "Extra CI" label to the PR triggers the cron build. This means that if you are modifying something which affects the builds we only cron you can test it on the PR before merging.

---
. o O (We could also do this semi-autonomously for online tests?!)

Reply:
@Cadair I can add the "Extra CI" label to trigger the cron test and publish test in a PR. Do you want to only run the online tests in a PR if an "Online CI" tag is added? (But always run online tests otherwise, i.e. merging PRs, tags and weekly cron.)

Reply:
I think we should keep the online build as default at the moment, and think about if that's something we can do in a bit more of a smart way.

Reply:
Running devdeps on the every PR seems like a waste of :bike: to me, especially with all the compiling.

Reply:
Yeah, it takes ~9 min to run currently. Maybe we should be using astropy's nightly PyPI repository? And for things which don't have that we could have a daily workflow which builds and caches wheels. Maybe just keep the cron stage as-is for now?